text_localization_environment/TextLocEnv.py

```diff
@@ -1,65 +1,82 @@
         :param true_bboxes: The true bounding boxes for each image
         :type image_paths: String or list
         :type true_bboxes: numpy.ndarray
         """
-        self.action_space = spaces.Discrete(9)
-        self.action_set = {0: self.right,
-                           1: self.left,
-                           2: self.up,
-                           3: self.down,
-                           4: self.bigger,
-                           5: self.smaller,
-                           6: self.fatter,
-                           7: self.taller,
-                           8: self.trigger
-                           }
-        # 224*224*3 (RGB image) + 9 * 10 (on-hot-enconded history) = 150618
-        self.observation_space = spaces.Tuple([spaces.Box(low=0, high=256, shape=(224,224,3)), spaces.Box(low=0,high=1,shape=(10,9))])
-        if type(image_paths) is not list: image_paths = [image_paths]
-        self.image_paths = image_paths
-        self.true_bboxes = [[TextLocEnv.to_standard_box(b) for b in bboxes] for bboxes in true_bboxes]
         # Determines whether the agent is training or testing
         # Optimizations can be applied during training that are not allowed for testing
         self.mode = mode
+        # Factor for scaling all bounding boxes relative to their size
+        self.bbox_scaling = bbox_scaling
         # Whether IoR markers will be placed upfront after loading the image
         self.premasking = premasking
         # Whether an episode terminates after a single trigger or is played out until the end
         self.playout_episode = playout_episode
         # Episodes will be terminated automatically after reaching max steps
         self.max_steps_per_image = max_steps_per_image
+        # Whether a termination action should be provided in the action set
+        self.has_termination_action = has_termination_action
+        # The type of IoR marker to be used when masking trigger regions
+        self.ior_marker_type = ior_marker_type
+
+        # Initialize action space
+        self.bbox_transformer = create_bbox_transformer(bbox_transformer)
+        self.action_space = spaces.Discrete(len(self.action_set))
+        # 224*224*3 (RGB image) + 9 * 10 (on-hot-enconded history) = 150618
+        self.observation_space = spaces.Tuple([
+            spaces.Box(low=0, high=256, shape=(224, 224, 3)),
+            spaces.Box(low=0, high=1, shape=(self.HISTORY_LENGTH, len(self.action_set)))
+        ])
+
+        # Initialize dataset
+        if type(image_paths) is not list:
+            image_paths = [image_paths]
+        self.image_paths = image_paths
+        self.true_bboxes = [[TextLocEnv.to_standard_box(b) for b in bboxes] for bboxes in true_bboxes]
+
+        # For registering a handler that will be executed once after a step
+        self.post_step_handler = None
+
+        # Episode-specific
 
         # Image for the current episode
         self.episode_image = Image.new("RGB", (256, 256))
```

Comment:
Why is this not 224x224?

Reply:
Doesn't really matter since it's just a place holder anyway